src/Microsoft.Health.Dicom.Api/Features/BackgroundServices/StartBlobDeleteMigrationService.cs

```diff
@@ -0,0 +1,109 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Features.Operations;
+using Microsoft.Health.Dicom.Core.Models.BlobMigration;
+using Microsoft.Health.Dicom.Core.Models.Operations;
+using Microsoft.Health.Operations;
+
+namespace Microsoft.Health.Dicom.Api.Features.BackgroundServices;
+
+public class StartBlobDeleteMigrationService : BackgroundService
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly BlobMigrationFormatType _blobMigrationFormatType;
+    private readonly bool _startBlobDelete;
+    private readonly ILogger<StartBlobDeleteMigrationService> _logger;
+    private readonly Guid _deleteOperationId;
+    private readonly Guid _copyOperationId;
+
+    public StartBlobDeleteMigrationService(
+        IServiceProvider serviceProvider,
+        IOptions<BlobMigrationConfiguration> blobMigrationFormatConfiguration,
+        ILogger<StartBlobDeleteMigrationService> logger)
+    {
+        EnsureArg.IsNotNull(serviceProvider, nameof(serviceProvider));
+        EnsureArg.IsNotNull(blobMigrationFormatConfiguration, nameof(blobMigrationFormatConfiguration));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _serviceProvider = serviceProvider;
+        _blobMigrationFormatType = blobMigrationFormatConfiguration.Value.FormatType;
+        _startBlobDelete = blobMigrationFormatConfiguration.Value.StartCopy;
```

Comment:
Perhaps we should just save the options as a member if we're saving 4 values from it?

Reply:
Done